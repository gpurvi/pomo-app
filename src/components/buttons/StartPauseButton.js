import React from 'react';
import {faPlay, faPause} from '@fortawesome/free-solid-svg-icons';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import '../../styles/components/TimerButton.css';

const StartPauseButton = ({startPauseClickHandler, timerStarted, breakTimerStarted}) => (

    <i
        onClick={startPauseClickHandler}
        className='d-inline-block mr-3 mr-lg-4'
    >
        {!timerStarted ? (
            breakTimerStarted ? (
                // grey play button
                <FontAwesomeIcon
                    size='5x'
                    className='timer-button--grey'
                    icon={faPlay}
                />
            ) : (
                //normal play button
                <FontAwesomeIcon size='5x' icon={faPlay}/>
            )
        ) : (
            // pause
            <FontAwesomeIcon size='5x' icon={faPause}/>
        )}
    </i>
);

export default StartPauseButton;