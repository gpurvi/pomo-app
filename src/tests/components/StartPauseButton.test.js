import React from 'react';
import {shallow} from 'enzyme';
import StartPauseButton from '../../components/buttons/StartPauseButton';

test('renders StartPauseButton with play image', () => {
    const wrapper = shallow(<StartPauseButton timerRunning={false}/>);
    expect(wrapper).toMatchSnapshot();
});

test('renders StopButton with pause image', () => {
    const wrapper = shallow(<StartPauseButton timerRunning={true}/>);
    expect(wrapper).toMatchSnapshot();
});

test('when clicked calls startPauseClickHandler', () => {
    const startPauseClickHandler = jest.fn();
    const wrapper = shallow(<StartPauseButton startPauseClickHandler={startPauseClickHandler}/>);
    wrapper.simulate('click');
    expect(startPauseClickHandler).toHaveBeenCalled();
});
