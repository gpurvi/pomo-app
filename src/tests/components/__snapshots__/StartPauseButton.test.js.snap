// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StartPauseButton with play image 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartPauseButton
    timerRunning={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <svg
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
>
        <path
                d="M8 5v14l11-7z"
        />
        <path
                d="M0 0h24v24H0z"
                fill="none"
        />
</svg>,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <path
            d="M8 5v14l11-7z"
/>,
          <path
            d="M0 0h24v24H0z"
            fill="none"
/>,
        ],
        "height": "48",
        "viewBox": "0 0 24 24",
        "width": "48",
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M8 5v14l11-7z",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M0 0h24v24H0z",
            "fill": "none",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
      ],
      "type": "svg",
    },
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          height="48"
          viewBox="0 0 24 24"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
>
          <path
                    d="M8 5v14l11-7z"
          />
          <path
                    d="M0 0h24v24H0z"
                    fill="none"
          />
</svg>,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <path
              d="M8 5v14l11-7z"
/>,
            <path
              d="M0 0h24v24H0z"
              fill="none"
/>,
          ],
          "height": "48",
          "viewBox": "0 0 24 24",
          "width": "48",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M8 5v14l11-7z",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M0 0h24v24H0z",
              "fill": "none",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
        ],
        "type": "svg",
      },
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders StopButton with pause image 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartPauseButton
    timerRunning={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <svg
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
>
        <path
                d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
        />
        <path
                d="M0 0h24v24H0z"
                fill="none"
        />
</svg>,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <path
            d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
/>,
          <path
            d="M0 0h24v24H0z"
            fill="none"
/>,
        ],
        "height": "48",
        "viewBox": "0 0 24 24",
        "width": "48",
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M6 19h4V5H6v14zm8-14v14h4V5h-4z",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M0 0h24v24H0z",
            "fill": "none",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
      ],
      "type": "svg",
    },
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          height="48"
          viewBox="0 0 24 24"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
>
          <path
                    d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
          />
          <path
                    d="M0 0h24v24H0z"
                    fill="none"
          />
</svg>,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <path
              d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
/>,
            <path
              d="M0 0h24v24H0z"
              fill="none"
/>,
          ],
          "height": "48",
          "viewBox": "0 0 24 24",
          "width": "48",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M6 19h4V5H6v14zm8-14v14h4V5h-4z",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M0 0h24v24H0z",
              "fill": "none",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
        ],
        "type": "svg",
      },
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
